src/averageAges.js

```diff
@@ -37,7 +43,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const haveChildren = people
+    .map(person => person.mother)
+    .filter(name => name);
+  const filteredWoman = people.filter(person => (!withChildren
+    || haveChildren.includes(person.name))
```

Comment:
Here we have O**2 complexity. We can avoid it. Just to iteration with reduce.
Task about learning different method, therefore it is nit required. But reduce make everything in a couple iteration))
```
  const mothers = people.reduce((acc, person) => {
    acc[person.mother] = person.name;

    return acc;
  }, {});

  const { womenSumAge, womenCount } = people.reduce((acc, person) => {
    const { name, sex, died, born } = person;

    const isPerson = withChildren ? mothers[name] : sex === 'f';

    return isPerson ? {
      womenSumAge: acc.womenSumAge + died - born,
      womenCount: acc.womenCount + 1,
    } : acc;
  }, {
    womenSumAge: 0, womenCount: 0,
  });

  return womenSumAge / womenCount;

```